packages/framework/src/handler.ts

```diff
@@ -176,8 +176,21 @@ export class NovuRequestHandler<Input extends any[] = any[], Output = any> {
     action: string
   ): Record<PostActionEnum, () => Promise<IActionResponse>> {
     return {
-      [PostActionEnum.TRIGGER]: this.triggerAction({ workflowId, ...body }),
+      [PostActionEnum.TRIGGER]: async () => {
+        const errors = this.client.getErrors();
+
+        if (errors.length !== 0) {
+          return this.createResponse(HttpStatusEnum.UNPROCESSABLE_ENTITY, { errors });
+        }
+
+        return this.triggerAction({ workflowId, ...body })();
+      },
       [PostActionEnum.EXECUTE]: async () => {
+        const errors = this.client.getErrors();
+
+        if (errors.length !== 0) {
```

Comment:
Let's generalize error handling across all action types to make the code dryer. Every byte counts in the framework SDK.